lib/features/rules/BpmnRules.js

```diff
@@ -144,7 +144,7 @@ BpmnRules.prototype.init = function() {
 
     return every(elements, function(element) {
       if (isConnection(element)) {
-        return canConnect(element.source, element.target, element);
+        return (canConnect(element.source, element.target, element) || {}).type === element.type;
```

Comment:
Let's elaborate what this does.

My guess is it ensures that connections can only be created if they would not change their type?

Reply:
Yes, the check was useless before because we want to create a number of shapes and connections that are predefined and must not change their type during creation.